spec/requests/session_requests_spec.rb

```diff
@@ -12,24 +12,46 @@
 
   describe "POST /sessions" do
     before(:all) do
-      @user = User.new(email: "hello@world.com", password: "Password1!")
+      @user = User.new(email: "hello@world.com", password: "Password1!", id: 123)
+      @user.save
     end
 
     it "creates a new session with valid email and password" do
-      post sessions_path, params: {user: {email: "hello@world.com", password: "Password1!"}}
+      post sessions_path, :params => { email: "hello@world.com", password: "Password1!" }  
       expect(session[:user_id]).to eq(@user.id)
     end
 
     it "fails to create a new session with invalid email" do
-      user = User.new(email: "hello@world.com", password: "Password1!")
-      post sessions_path, params: {user: {email: "notanemail", password: "Password1!"}}
+      post sessions_path, :params => { email: "notanemail", password: "Password1!" }
       expect(session[:user_id]).to eq(nil)
     end
 
     it "fails to create a new session with invalid password" do
-      user = User.new(email: "hello@world.com", password: "Password1!")
-      post sessions_path, params: {user: {email: "hello@world.com", password: nil}}
+      post sessions_path, :params => { email: "hello@world.com", password: nil }
       expect(session[:user_id]).to eq(nil)
     end
+
+    after(:all) do
+      @user.delete
```

Comment:
You shouldn't need to do this. 
You don't want to share the same object across multiple tests

Reply:
No, we don't want the same object to persist across multiple tests. It will make the tests brittle and unreliable. 
Each `it` block should have a clean state. Try using `let`